### ISSUE TITLE
Various bugs in heart_prediction model.

### ISSUE THREAD

**Issue by yashashwanidixit** [Niketkumardheeryan/ML-CaPsule#1310]
**Describe the bug**
There are several bugs in this model:
1.The model instantiation and score calculation : lr = LogisticRegression(solver='liblinear') and score=cross_val_score(lr,x,y)  uses unscaled parameter (x) due to which the logistic regression model is stopping before it could find the best possible mathematical solution(or it is not converging ) .I tried to increase the iterations to 1000 and used liblinear solver but still it is not converging.
2. The UCI heart disease kaggle dataset link given in the readme is errornous.
3. The train_test_split imported in cell 114 is never actually used.
4. This is not a bug but the accuracy of the model can be improved by incorporating gridsearchcv.
**To Reproduce**
Steps to reproduce the behavior
Run the cell below 128th cell containing the code:lr = LogisticRegression()
score=cross_val_score(lr,x,y) 

**Solution**:
1. Scale x through standardscaler.fittransform() method and used the scaled variable instead in cross entropy calculation.
2. FIx the readme file error .
3. Inlcude gridsearchcv for logistic regression or decision tress models.

**Screenshots**

<img width="739" height="765" alt="Image" src="https://github.com/user-attachments/assets/d69c3893-0ad9-42cc-bc29-0897852a7db2" />

**Desktop (please complete the following information):**
I ran the code on kaggle.

I am a gssoc "26 ai/agents_track contributor and would like to work on all these mentioned issues  and furthur increase the accuracy of this model.Please assign this issue to me under GSSOC '26.
Thankyou so much.

**Comment by nikhhiilsingghvi** [comment]
I am a gssoc "26 ai/agents_track contributor and would like to work on all these mentioned issues and furthur increase the accuracy of this model.Please assign this issue to me under GSSOC '26.
Thankyou so much.
